calliope/backend/checks.py

```diff
@@ -32,202 +61,56 @@ def check_operate_params(model_data):
         serious issues that should raise a ModelError
 
     """
-    defaults = UpdateObserverDict(
-        initial_yaml_string=model_data.attrs["defaults"],
-        name="defaults",
-        observer=model_data,
-        flat=True,
-    )
-    run_config = UpdateObserverDict(
-        initial_yaml_string=model_data.attrs["run_config"],
-        name="run_config",
-        observer=model_data,
-    )
-
-    warnings, errors = [], []
-    comments = AttrDict()
-
-    def _is_missing(var):
-        if var not in model_data.data_vars.keys():
-            return True
+    checklist = AttrDict.from_yaml(checklist_path)
+
+    logs = {"warning": [], "error": []}
+
+    for check_name, check_config in checklist.items():
+        imask = imask_where(model_data, check_name, check_config.where)
+        if isinstance(imask, xr.DataArray) and imask.any() or imask is True:
+            if "assert" in check_config.keys():
+                imask *= ~_equation_parser(model_data, check_config["assert"])
+            if isinstance(imask, xr.DataArray) and imask.any() or imask is True:
+                for exctype, exclist in logs.items():
+                    if exctype in check_config.keys():
+                        exclist.append(check_config[exctype])
+                if "must_exist" in check_config.keys():
+                    _create_var(check_config["must_exist"], model_data, run_config)
+
+    return logs["warning"], logs["error"]
+
+
+def _create_var(var_to_create, model_data, run_config):
```

Comment:
For functions like this that look pretty impenetrable I think we should take the minimal effort of writing a docstring that says what the function's parameters are and what it returns